test/github-issues/4096/issue-4096.ts

```diff
@@ -0,0 +1,61 @@
+import "reflect-metadata";
+import {createTestingConnections, closeTestingConnections, reloadTestingDatabases} from "../../utils/test-utils";
+import {Connection} from "../../../src/connection/Connection";
+import {expect} from "chai";
+import {User} from "./entity/User";
+
+describe("github issues > #4096 SQLite support for orUpdate", () => {
+    let connections: Connection[];
+
+    before(async () => connections = await createTestingConnections({
+        entities: [User],
+        enabledDrivers: ["sqlite"],
+        schemaCreate: true,
+        dropSchema: true,
+    }));
+
+    beforeEach(() => reloadTestingDatabases(connections));
+
+    after(() => closeTestingConnections(connections));
+
+    const user1 = new User();
```

Comment:
Object isn't shared between tests in any way, so we could put it in `it()` block.

Reply:
Moved 👌